### PR TITLE
migration pattern

### PR DESCRIPTION
**Pull request by dpensi**
unit tests

# Description
added migration support

Fixes CCXDEV-8774

## Type of change

Please delete options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Testing steps

N/A

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: ccx_notification_writer.go [hunk 1]
```diff
@@ -474,3 +484,57 @@ func main() {
...
+	}
+
+	log.Info().Msgf("Current DB version: %d", currMigVer)
+	log.Info().Msgf("Maximum available version: %d", utils.GetMaxVersion())
+	return ExitStatusOK, nil
+}
+
+// PerformMigrations migrates the database to the version
+// specified in params
+func PerformMigrations(conf ConfigStruct, migParam string) (int, error) {
```
Comment (tisnik) [inline ccx_notification_writer.go:509]: I'd name the returned `int`

File: mig_0001_create_events_targets_tbl.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	types "github.com/RedHatInsights/insights-results-types"
+)
+
+var mig0001CreateEventTargetsTbl = mig.Migration{
+	StepUp: func(tx *sql.Tx, _ types.DBDriver) error {
+		_, err := tx.Exec(`
+			CREATE TABLE event_targets (
+				id				INTEGER NOT NULL,
+				name          	VARCHAR NOT NULL,
+				metainfo        VARCHAR NOT NULL,
```
Comment (tisnik) [inline mig_0001_create_events_targets_tbl.go:30]: missing space :)

Comment (tisnik) [reply mig_0001_create_events_targets_tbl.go:30]: also name and metainfo should be unique (constraint)

**Review by tisnik — COMMENTED** [review]
LGTM in overall, just some comments

File: mig_0001_create_events_targets_tbl.go (hunk 2 above)
Comment (dpensi) [reply mig_0001_create_events_targets_tbl.go]: the space it's like this only on git hub, in the edito is fine

Comment (tisnik) [reply mig_0001_create_events_targets_tbl.go]: there's tab inside the string, which might be problematic later, let's see (still minor thing)

File: ccx_notification_writer.go (hunk 1 above)
Comment (tisnik) [reply ccx_notification_writer.go]: cool

File: mig_0001_create_events_targets_tbl.go (hunk 2 above)
Comment (dpensi) [reply mig_0001_create_events_targets_tbl.go]: right! fixed, thx

**Review by tisnik — APPROVED** [review]
LGTM

**Review by Bee-lee — APPROVED** [review]
looks good

**Pull request merged**